src/App.jsx

```diff
@@ -41,6 +40,13 @@ function App() {
         latitude: location.lat,
       };
 
+      if (
+        coordinates.latitude === geoData.latitude &&
+        coordinates.longitude === geoData.longitude
+      ) {
+        setLoading(false);
+      }
```

Comment:
If the user enters the same city twice, it would cause an infinite loop with the animation rendering. This checks if the coordinates of the new city is the same as the previous coordinates and if so, the animation is NOT triggered.

Reply:
I think this is happening because on initial load, the weather displayed is dependent on `geoData`, while subsequent weather API calls are based on `searchInput`.  

Since `searchInput` is initially empty, entering `portland` will trigger another `retrieveGeoData()` call even if Portland's weather is being displayed.

Reply:
Oh yeah you're right, I didn't notice that. The loading animation only occurs if the retrieveGeoData function is called. Seems like the only way to resolve is to either remove the default long/lat we set for geoData so that the application doesn't display anything on load, or move the animation into WeatherProvider component. I don't think the latter would work because I need access to the parent div to add the opacity affect on the screen during the animation.

Reply:
Another thing I noticed was if you search for lets say New York , the animation works. Then if you search with no input, then New York again, the animation loads for a quick second. Which is odd because there is a conditional that checks if the input is empty, we don't render the component (line 22) so I'm not sure why this occurs.

Reply:
I think it's nice having a location loaded by default, and it looks good for demonstration purposes.  

Regarding the last comment, you were right, that's because a blank search is being replaced by the city name again.